### ISSUE TITLE
'/' filename separators work improperly under Windows

### ISSUE THREAD

**Issue by Cipher731** [markokr/rarfile#91]
The error is like `rarfile.BadRarFile: Failed the read enough data: req=123456 got=0`. 

Notice the "got=0", which means UnRAR.exe extracts nothing.

I find out that the problem is windows version of UnRAR.exe doesn't regard '/' as path separators in path parameter.

Downgrade to 3.3 solves this problem.

**Comment by RandomNameUser** [comment]
The fix doesn't seem to work for me, 3.3 has the same problem, just for a different reason. 

3.3 is more correct, in that in `CommonParser._open_unrar` it does the following:

```
            if PATH_SEP != os.sep:
                fn = fn.replace(PATH_SEP, os.sep)
``` 

(which is missing in 4.0). Unfortunately `ToolSetup.open_cmdline` adds the filename with `add_file_arg`, which reverses the change. For testing I replaced that with `cmdline.append(filefn)` and now it works for me, but I hate using patched libs (that's one of the major strengths of Python. having all those great libs available).

But it really should be fixed in the 4.x branch, this is a pretty major issue, I would think.

**Comment by markokr** [comment]
Should be fixed by #96
